example/pubspec.lock

```diff
@@ -80,14 +80,21 @@ packages:
       path: ".."
       relative: true
     source: path
-    version: "0.0.1"
+    version: "0.3.0+1"
   matcher:
     dependency: transitive
     description:
       name: matcher
       url: "https://pub.dartlang.org"
     source: hosted
-    version: "0.12.10"
+    version: "0.12.11"
+  material_color_utilities:
+    dependency: transitive
+    description:
+      name: material_color_utilities
+      url: "https://pub.dartlang.org"
+    source: hosted
+    version: "0.1.3"
```

Comment:
I don't understand how there are changes in the `pubspec.lock` but there are no changes in the `pubspec.yaml`, WDYT ?

Or it could be that you missed it in the PR by mistake, or if you made a change to the example to improve it, it would be great if you did it in another PR.

Reply:
I ran `flutter pub get` in the root of the project and these packages were implicitly updated. I didn't change anything in the example project.

```
 jonathantrowbridge@Jonathans-MacBook-Pro-2  ~/projects/iterable-flutter   master  git status
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
 jonathantrowbridge@Jonathans-MacBook-Pro-2  ~/projects/iterable-flutter   master  flutter pub get
Running "flutter pub get" in iterable-flutter...                   909ms
Running "flutter pub get" in example...                          1,089ms
 jonathantrowbridge@Jonathans-MacBook-Pro-2  ~/projects/iterable-flutter   master ●  git status
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   example/pubspec.lock
        modified:   pubspec.lock

no changes added to commit (use "git add" and/or "git commit -a")
```

Reply:
I just searched around a bit and it looks like dart.dev recommends against committing pubspec.dart for library packages. More information:

From the documentation:
 
>pubspec.lock
>
>The pubspec.lock file is a special case, similar to Ruby’s Gemfile.lock.
>
>For library packages, don’t commit the pubspec.lock file. Regenerating the pubspec.lock file lets you test your >package against the latest compatible versions of its dependencies.
>
>For application packages, we recommend that you commit the pubspec.lock file. Saving pubspec.lock ensures that everyone working on the app uses the exact same versions._


- https://dart.dev/guides/libraries/private-files#pubspeclock
- https://stackoverflow.com/a/64750701/563509

Reply:
@Santi92 how would you like me to handle the pubspec.lock file? Once we get this merged down I have another PR I'd like to submit for review. This PR adds support for tracking event data on iOS and Android: https://github.com/masterwok/iterable-flutter/pull/1